src/main/java/net/snowflake/ingest/connection/ServiceResponseHandler.java

```diff
@@ -128,49 +106,98 @@ public static HistoryRangeResponse unmarshallHistoryRangeResponse(HttpResponse r
+      throw new IllegalArgumentException();
     }
 
+    // handle the exceptional status code
+    handleExceptionalStatus(response);
+
     // grab the string version of the response entity
     String blob = EntityUtils.toString(response.getEntity());
 
     // read out our blob into a pojo
-    return mapper.readValue(blob, HistoryRangeResponse.class);
+    return mapper.readValue(blob, ConfigureClientResponse.class);
+  }
+
+  /**
+   * unmarshallGetClientStatus - Given an HttpResponse object, attempts to deserialize it into a
+   * ClientStatusResponse
+   *
+   * @param response HttpResponse
+   * @return ClientStatusResponse
+   * @throws IOException
+   * @throws IngestResponseException
+   */
+  public static ClientStatusResponse unmarshallGetClientStatus(HttpResponse response)
+      throws IOException, IngestResponseException {
+    if (response == null) {
+      LOGGER.warn("Null response passed to unmarshallClientStatusResponse");
+      throw new IllegalArgumentException();
+    }
+
+    // handle the exceptional status code
+    handleExceptionalStatus(response);
+
+    // grab the string version of the response entity
+    String blob = EntityUtils.toString(response.getEntity());
+
+    // read out our blob into a pojo
+    return mapper.readValue(blob, ClientStatusResponse.class);
   }
 
   /**
-   * handleExceptionStatusCode - throws the correct error for a status
+   * handleExceptionStatusCode - throws the correct error when response status is not OK
    *
-   * @param statusLine the status line we want to check
-   * @throws BackOffException -- if we have a 503 exception
-   * @throws IOException - if we don't know what it is
+   * @param response HttpResponse
+   * @throws BackOffException if we have a 503 exception
+   * @throws IOException if we don't know what it is
    */
-  private static void handleExceptionalStatus(StatusLine statusLine, HttpResponse response)
+  private static void handleExceptionalStatus(HttpResponse response)
       throws IOException, IngestResponseException {
-    // if we have a 503 exception throw a backoff
-    switch (statusLine.getStatusCode()) {
-        // If we have a 503, BACKOFF
-      case HttpStatus.SC_SERVICE_UNAVAILABLE:
-        LOGGER.warn("503 Status hit, backoff");
-        throw new BackOffException();
-
-        // We don't know how to respond now...
-      default:
-        LOGGER.error("Status code {} found in response from service", statusLine.getStatusCode());
-        String blob = EntityUtils.toString(response.getEntity());
-        throw new IngestResponseException(
-            statusLine.getStatusCode(),
-            IngestResponseException.IngestExceptionBody.parseBody(blob));
+    StatusLine statusLine = response.getStatusLine();
+    if (!isStatusOK(statusLine)) {
+      // A network issue occurred!
+      LOGGER.warn(
+          "Exceptional Status Code found in " + "unmarshallHistoryRangeResponse - {}",
```

Comment:
nit: Since this is used by all responses. should we remove `unmarshallHistoryRangeResponse`? and may be pass from caller?